src/status_im2/subs/shell.cljs

```diff
@@ -190,19 +190,19 @@
    (let [{:keys [chats-stack community-stack]}
          (reduce
           (fn [acc [_ {:keys [unviewed-messages-count unviewed-mentions-count chat-type muted]}]]
-            (case chat-type
-              constants/community-chat-type
-              (when-not muted
-                (-> acc
-                    (update-in [:community-stack :unviewed-messages-count] + unviewed-messages-count)
-                    (update-in [:community-stack :unviewed-mentions-count] + unviewed-mentions-count)))
-
-              (constants/private-group-chat-type constants/one-to-one-chat-type)
-              (when-not muted
-                (-> acc
-                    (update-in [:chats-stack :unviewed-messages-count] + unviewed-messages-count)
-                    (update-in [:chats-stack :unviewed-mentions-count] + unviewed-mentions-count)))
-
+            (cond
+              (and (not muted)
+                   (= chat-type constants/community-chat-type))
+              (-> acc
+                  (update-in [:community-stack :unviewed-messages-count] + unviewed-messages-count)
+                  (update-in [:community-stack :unviewed-mentions-count] + unviewed-mentions-count))
+
+              (and (not muted)
+                   (#{constants/private-group-chat-type constants/one-to-one-chat-type} chat-type))
+              (-> acc
+                  (update-in [:chats-stack :unviewed-messages-count] + unviewed-messages-count)
+                  (update-in [:chats-stack :unviewed-mentions-count] + unviewed-mentions-count))
+              :else
```

Comment:
The `cond` is better than `case`, Because if `case` failed it'll return `nil` because of the `(when)` function call.
But cond on the other hand will always return the accumulator.

The issue here was that the notification indicator was failing if it found one muted chat, It returned `nil` as the value and that meant no notification dot at all.